src/olympia/promoted/models.py

```diff
@@ -80,16 +79,26 @@ def approve_for_version(self, version):
                 group_id=self.group_id,
                 application_id=app.id)
 
+    def approve_for_addon(self):
+        """This sets up the addon as approved for the current promoted group.
+
+        The current version will be signed for approval, and if there's special
+        signing needed for that group the version will be resigned."""
+        from olympia.lib.crypto.tasks import sign_addons
+
+        self.approve_for_version(self.addon.current_version)
+        if self.group.autograph_signing_states:
+            sign_addons([self.addon.id], send_emails=False)
+
     def save(self, *args, **kwargs):
         super().save(*args, **kwargs)
 
-        if (
-            self.group in PROMOTED_GROUPS_FOR_SUBSCRIPTION and
-            not PromotedSubscription.objects.filter(
-                promoted_addon=self
-            ).exists()
-        ):
-            PromotedSubscription.objects.create(promoted_addon=self)
+        if self.group.subscription:
+            if not hasattr(self, 'promotedsubscription'):
+                PromotedSubscription.objects.create(promoted_addon=self)
+        elif (self.group.immediate_approval and
+              self.approved_applications != self.all_applications):
+            self.approve_for_addon()
```

Comment:
Is this backward compatible with the current -prod behavior? (i.e. the subscription feature is only enabled on -dev)

Reply:
The changes I made would be backwards compatible - it only wouldn't be if `SPONSORED` or `VERIFIED` were `immediate_approval = True` but it's only `SPOTLIGHT` (currently).

That said, looking again at the existing code you landed a few days ago - that would create `PromotedSubscription` instances if an existing `VERIFIED` or `SPONSORED` was edited/added regardless of the waffle state.  Is that a problem?  (I guess not because we don't check for promotedsubscription anywhere currently)

Reply:
>  Is that a problem? (I guess not because we don't check for promotedsubscription anywhere currently)

I don't think it's a problem because we cannot do anything without the waffle enabled.